### PR TITLE
Fix #378: Update hint text behavior for text, numeric, and fractions input to match Oppia web.

### LINKED ISSUE #378: Text-based interactions need correct hint text
We should use the same hint text as Oppia web for text-based interactions (e.g. 'Enter a fraction in the form x/x').

### PR DESCRIPTION
**Pull request by BenHenning**
Fix #378.

This PR updates the hint text behavior for text, numeric, and fractions input interaction views to match Oppia web (see #378 for details). In general, this means:
- Text input does not use a hint text unless one is specified
- Fractions input uses either a custom hint text, or one of two defaults based on whether whole numbers are allowed
- Numeric input does not show hint text

Tests are not added due to #388 and the inability to thoroughly test these interactions at the state level.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/player/state/itemviewmodel/FractionInteractionViewModel.kt [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
 import org.oppia.app.parser.StringToFractionParser
 import org.oppia.app.player.state.answerhandling.InteractionAnswerHandler
 import org.oppia.domain.util.toAnswerString
 
+/** View model corresponding to fraction interactions. */
 class FractionInteractionViewModel(
-  existingAnswer: InteractionObject?, val isReadOnly: Boolean
+  interaction: Interaction, existingAnswer: InteractionObject?, val isReadOnly: Boolean,
+  private val context: Context
 ): StateItemViewModel(), InteractionAnswerHandler {
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/player/state/itemviewmodel/FractionInteractionViewModel.kt]: Nit: space issue: `) : StateItemViewModel`

File: app/src/main/java/org/oppia/app/player/state/itemviewmodel/TextInputViewModel.kt [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
+import org.oppia.app.model.Interaction
 import org.oppia.app.model.InteractionObject
 import org.oppia.app.player.state.answerhandling.InteractionAnswerHandler
 import org.oppia.domain.util.toAnswerString
 
+/** View model for text input interaction views. */
 class TextInputViewModel(
-  existingAnswer: InteractionObject?, val isReadOnly: Boolean
+  interaction: Interaction, existingAnswer: InteractionObject?, val isReadOnly: Boolean
 ): StateItemViewModel(), InteractionAnswerHandler {
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/player/state/itemviewmodel/TextInputViewModel.kt]: Nit: space issue `) : StateItemViewModel()`

**Review by rt4914 — APPROVED** [review]
LGTM

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -38,10 +38,9 @@
   <string name="topic_play_chapter_index">%s.</string>
   <string name="chapter_name">Chapter %s: %s</string>
   <string name="chapter_completed">Chapter %s: %s is completed.</string>
-  <string name="fractions_hint_text">Write fraction here.</string>
-  <string name="number_input_hint_text">Write numbers here.</string>
+  <string name="fractions_default_hint_text">Enter a fraction in the form x/x, or a mixed number in the form x x/x.</string>
```
Comment (nikitamarysolomanpvt) [inline app/src/main/res/values/strings.xml:43]: I was using the error message as same as that of web ...PTAL on  #419

File: app/src/main/java/org/oppia/app/customview/interaction/NumericInputInteractionView.kt [hunk 4]
```diff
@@ -18,19 +18,4 @@ class NumericInputInteractionView @JvmOverloads constructor(
   context: Context,
   attrs: AttributeSet? = null,
   defStyle: Int = android.R.attr.editTextStyle
-) : EditText(context, attrs, defStyle), View.OnFocusChangeListener {
```
Comment (nikitamarysolomanpvt) [inline app/src/main/java/org/oppia/app/customview/interaction/NumericInputInteractionView.kt:26]: This section is removed  or due to merge conflict it looks like this?
In NumericInputInteractionView there wont be default hint as well? Last time in accessibility test we were getting suggestion of label missing , when we are not showing hint on focus... Does  this new change means there wont be hint while input view is not in focus...?

**Review by nikitamarysolomanpvt — COMMENTED** [review]
PTAL on my comments ... beside that LGTM

File: app/src/main/java/org/oppia/app/player/state/itemviewmodel/FractionInteractionViewModel.kt (hunk 1 above)
Comment (BenHenning) [reply app/src/main/java/org/oppia/app/player/state/itemviewmodel/FractionInteractionViewModel.kt]: Done.

File: app/src/main/java/org/oppia/app/player/state/itemviewmodel/TextInputViewModel.kt (hunk 2 above)
Comment (BenHenning) [reply app/src/main/java/org/oppia/app/player/state/itemviewmodel/TextInputViewModel.kt]: Done, and elsewhere.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (BenHenning) [reply app/src/main/res/values/strings.xml:43]: Sorry, I'm not sure I see how #419 relates to this PR. This PR is meant to change the non-error hint text to match Oppia web.

File: app/src/main/java/org/oppia/app/customview/interaction/NumericInputInteractionView.kt (hunk 4 above)
Comment (BenHenning) [reply app/src/main/java/org/oppia/app/customview/interaction/NumericInputInteractionView.kt:26]: There is no input for numeric input on Oppia web, but you're definitely correct that we need to have a content description. Updated numeric input to set a content description, and text input to use a default content description when no hint is set. Verified these work as expected with TalkBack.

Comment (BenHenning) [reply app/src/main/java/org/oppia/app/customview/interaction/NumericInputInteractionView.kt:26]: Also: yes, the focus stuff was removed on accident. However, the new code seems to be working as expected.

**Comment by BenHenning** [comment]
Thanks!

**Pull request merged**